include/seqan3/argument_parser/detail/version_check.hpp

```diff
@@ -0,0 +1,415 @@
+     *    * If the current app version is greater than the one returned by the server call, we assume that the
+     *      developer has released a new version and is notified to send us the new version
+     *      (see seqan3::version_checker::MESSAGE_REGISTERED_APP_UPDATE).
+     *    **Release mode** (directed at the user of the application):
+     *    * If the current app version is lower than the one returned by the server call, the user is notified that
+     *      a newer version exists.
+     */
+    void operator()(std::promise<bool> prom)
+    {
+        if (!std::filesystem::exists(cookie_path))
+        {
+            prom.set_value(false);
+            return;
+        }
+
+        std::ifstream timestamp_file{timestamp_filename};
+        std::string cookie_line{};
+
+        if (timestamp_file.is_open())
+        {
+            std::getline(timestamp_file, cookie_line);
+            timestamp_file.close();
+
+            if (cookie_line == "NEVER")
+            {
+                prom.set_value(false);
+                return;
+            }
+            else if (cookie_line != "ALWAYS") // then it is a timestamp
+            {
+                double file_time_diff{get_time_diff_to_current(cookie_line)}; // time difference in seconds
+
+                if (file_time_diff < 86400/*one day in seconds*/)
+                {
+                    prom.set_value(false); // only check for newer version once a day
+                    return;
+                }
+            }
+        }
+
+        std::array<int, 3> empty_version{0, 0, 0};
+        std::array<int, 3> srv_app_version{};
+        std::array<int, 3> srv_seqan_version{};
+
+        std::ifstream version_file{cookie_path / (name + ".version")};
+
+        if (version_file.is_open())
+        {
+            std::string line{};
+            std::getline(version_file, line); // get first line which should only contain the version number of the app
+
+        if (line != UNREGISTERED_APP)
+            srv_app_version = get_numbers_from_version_string(line);
+#if !defined(NDEBUG) || defined(SEQAN3_TEST_VERSION_CHECK_)
+        else
+            std::cerr << MESSAGE_UNREGISTERED_APP;
+#endif // !defined(NDEBUG) || defined(SEQAN3_TEST_VERSION_CHECK_)
+
+            std::getline(version_file, line); // get second line which should only contain the version number of seqan
+            srv_seqan_version = get_numbers_from_version_string(line);
+
+            version_file.close();
+        }
+
+#if !defined(NDEBUG) || defined(SEQAN3_TEST_VERSION_CHECK_) // only check seqan version in debug or testing mode
+        if (srv_seqan_version != empty_version)
+        {
+            std::array<int, 3> seqan_version = {SEQAN3_VERSION_MAJOR, SEQAN3_VERSION_MINOR, SEQAN3_VERSION_PATCH};
+
+            if (seqan_version < srv_seqan_version)
+                std::cerr << MESSAGE_SEQAN3_UPDATE;
```

Comment:
what's this?

Reply:
This prints "hey developer, There is a new seqan library version."

Reply:
why is it ALL_CAPS? It looks like a macro...